BridgeSDK/BridgeAPI/SBBBridgeNetworkManager.m

```diff
@@ -163,6 +164,26 @@ - (void)checkForAndHandleUnsupportedAppVersionHTTPError:(NSError *)error
     }
 }
 
+- (void)checkForAndHandleServerPreconditionNotMetHTTPError:(NSError *)error task:(NSURLSessionDataTask *)task responseObject:(id)responseObject retryObject:(APCNetworkRetryObject *)retryObject
+{
+    if (error.code == SBBErrorCodeServerPreconditionNotMet)
+    {
+        // Look to see if the app delegate handles this error.
+        // Note: check conforms to protocol to ensure that the app delegate is intentionally
+        // implementing this method and not coincidentally using the same method signature for something else.
+        id appDelegate = [[UIApplication sharedApplication] delegate];
+        if (![appDelegate conformsToProtocol:@protocol(SBBBridgeAppDelegate)] ||
+            ![appDelegate respondsToSelector:@selector(handleUserNotConsentedError:sessionInfo:networkManager:)] ||
+            ![appDelegate handleUserNotConsentedError:error sessionInfo:responseObject networkManager:self])
+        {
+#if DEBUG
+            // Log the error to the console
+            NSLog(@"User Not Consented error not handled by app delegate:\n%@", error);
+#endif
```

Comment:
There didn’t seem like a reasonable default way to handle it in the SDK as there is with 410-you-need-to-upgrade, since it requires study-specific UI to get the user to re-consent, so it just logs the error to the console and (in any case) ends up sending the error to the completion handler, which is where it’s currently handled by existing apps anyway.

> On Jan 12, 2016, at 2:49 PM, Shannon Young <notifications@github.com> wrote:
> 
> In BridgeSDK/BridgeAPI/SBBBridgeNetworkManager.m <https://github.com/Sage-Bionetworks/Bridge-iOS-SDK/pull/50#discussion_r49526940>:
> 
> > +- (void)checkForAndHandleServerPreconditionNotMetHTTPError:(NSError *)error task:(NSURLSessionDataTask *)task responseObject:(id)responseObject retryObject:(APCNetworkRetryObject *)retryObject
> > +{
> > +    if (error.code == SBBErrorCodeServerPreconditionNotMet)
> > +    {
> > +        // Look to see if the app delegate handles this error.
> > +        // Note: check conforms to protocol to ensure that the app delegate is intentionally
> > +        // implementing this method and not coincidentally using the same method signature for something else.
> > +        id appDelegate = [[UIApplication sharedApplication] delegate];
> > +        if (![appDelegate conformsToProtocol:@protocol(SBBBridgeAppDelegate)] ||
> > +            ![appDelegate respondsToSelector:@selector(handleUserNotConsentedError:sessionInfo:networkManager:)] ||
> > +            ![appDelegate handleUserNotConsentedError:error sessionInfo:responseObject networkManager:self])
> > +        {
> > +#if DEBUG
> > +            // Log the error to the console
> > +            NSLog(@"User Not Consented error not handled by app delegate:\n%@", error);
> > +#endif
> app version handler has a fall-through for the case where the app delegate does not handle it. What happens in this case?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/Sage-Bionetworks/Bridge-iOS-SDK/pull/50/files#r49526940>.
>